src/views/ChooseEdgeConnection.vue

```diff
@@ -0,0 +1,154 @@
+            </p>
+          </v-card-text>
+          <v-card-actions>
+            <v-row justify="center">
+              <v-btn
+                color="primary"
+                :small="$vuetify.breakpoint.mdAndUp"
+              >
+                <v-icon v-if="$vuetify.breakpoint.xsOnly">
+                  mdi-wifi
+                </v-icon>
+                <span v-if="$vuetify.breakpoint.smAndUp">
+                  Local Network
+                </span>
+              </v-btn>
+            </v-row>
+          </v-card-actions>
+        </v-card>
+      </v-col>
+      <!-- middle column -->
+      <v-col cols="12">
+        <v-row justify="center">
+          <v-card flat>
+            <v-card-title>
+              OR
+            </v-card-title>
+          </v-card>
+        </v-row>
+      </v-col>
+      <!-- third column -->
+      <v-card
+        max-width="500"
+        class="text-center"
+      >
+        <v-card-text>
+          <p class="text">
+            Connect to remote network
+          </p>
+          <v-subheader>
+            (must enter Peer ID to Ambianic network)
+          </v-subheader>
+          <v-text-field
+            v-model="edgeAddress"
+            type="text"
+            label="Peer ID to Ambianic Network*"
+            placeholder="Enter Peer ID"
+            outlined
+            dense
+            class="mt-4"
+          />
+        </v-card-text>
+
+        <v-card-actions>
+          <v-btn
+            :disabled="!correctEdgeAddress"
+            @click="sendEdgeAddress"
+            color="primary"
+          >
+            REMOTE NETWORK
+          </v-btn>
+        </v-card-actions>
+      </v-card>
+    </v-row>
+    <div
+      v-else
+    >
+      {{ this.$router.push('edge-connect') }}
+    </div>
+  </app-frame>
+</template>
+<script>
+import AppFrame from '@/components/AppFrame.vue'
+import {
+  REMOVE_REMOTE_PEER_ID
+} from '../store/action-types.js'
+import { mapState } from 'vuex'
+import {
+  PEER_CONNECTED
+} from '@/store/mutation-types'
+
+export default {
+  data: () => {
+    return {
+      edgeAddress: '',
+      correctEdgeAddress: false
+      // connectionStatus: '',
+      // connectionTip: '',
+      // testInProgress: false,
+      // testDone: true,
+      // statusColor: 'info',
+      // resetEdgeDialog: false
+    }
+  },
+  methods: {
+    // Validate the user input so the ID has the correct format before showing the connect button
+    validateIP (value) {
+      if (/^([a-zA-Z0-9]{8})-([a-zA-Z0-9]{4})-([a-zA-Z0-9]{4})-([a-zA-Z0-9]{4})-([a-zA-Z0-9]{12})$/.test(value)) {
+        this.correctEdgeAddress = true
+        return this.correctEdgeAddress
+      } else {
+        // if value is not matching regex, remove button
+        this.correctEdgeAddress = false
+        return this.correctEdgeAddress
+      }
+    },
+    sendEdgeAddress () {
+      console.log(this.edgeAddress)
+      this.$store.state.pnp.edgeRoom = this.edgeAddress
```

Comment:
@BKristenssonAlfsson I notice here that you are directly altering Vuex store state from component code. This can mess up vuex promises to manage state integrity for the application. Vuex requires all state changes to be managed via COMMIT (synchronous) or DISPATCH (async). See [here](https://vuex.vuejs.org/guide/).

> At the center of every Vuex application is the store. A "store" is basically a container that holds your application state. There are two things that make a Vuex store different from a plain global object:
> 
> Vuex stores are reactive. When Vue components retrieve state from it, they will reactively and efficiently update if the store's state changes.
>
> **You cannot directly mutate the store's state**. The only way to change a store's state is by explicitly committing mutations. This ensures every state change leaves a track-able record, and enables tooling that helps us better understand our applications.

Reply:
@sniperadmin you can correct me here, but I believe the way to implement switching from a known edge peer id to a new one is to use a DISPATCH call like this one:
```
this.$store.dispatch(UPDATE_REMOTE_PEER_ID)
```

which in turn will have an implementation body that looks something like this

```
this.$store.dispatch(REMOVE_REMOTE_PEER_ID)
this.$store.COMMIT(NEW_REMOTE_EDGE_PEER_ID)
```